### PR TITLE
Issue98 create project vault

### PR DESCRIPTION
**Pull request by gano3s**
Add vault functionality.

## Short description:
This enhancement provides a capability to safely store the project's related secret. Like discovered or obtained passwords, tokens or even generic note related to project.

## About implementation
I am using the openssl_encrypt/openssl_decrypt methods to encrypt the secret and store it in the database. The used algorithm is 'AES-256-CTR'.
I intentionaly allow to delete the secret without knowledge of password. It is because even during the development, I several times forgot the temporary password, which was used to the encryption.

Each item in vault contains flag if it should be reportable. If it is so, then the name and note about the secret are reported in the final report. I decided to not put the secret value itself into the report. It would cause to many issues: when to ask for the password? How to prevent anybody to access generated report with passwords? and similar. This way, we can report which accounts were compromised and if needed, the passwords can be extracted manually.

I tried to cover all controllers with the unit tests.

## Examples

https://user-images.githubusercontent.com/2551605/156604531-3114942c-bff0-4343-8dbc-4730fff05cef.mov

https://user-images.githubusercontent.com/2551605/156604587-a21c7d8c-ac0d-411e-ba57-ce3befe46a9a.mov

https://user-images.githubusercontent.com/2551605/156604640-1df35c3f-3345-47fa-9a23-b1b40786f96e.mov

## Related PRs
https://github.com/reconmap/rest-api/pull/39
https://github.com/reconmap/documentation/pull/12

### CODE REVIEW

**Comment by santiagolizardo** [comment]
Great job once again!

File: src/components/projects/vault/VaultItemEdit.js [hunk 1]
```diff
@@ -0,0 +1,127 @@
...
+            })
+    }
+
+    const onPasswordProvided = ev => {
+        ev.preventDefault();
+
+        secureApiFetch(`/vault/${projectId}/${vaultItemId}`, { method: 'POST', body: JSON.stringify({ 'password' : password }) })
+            .then(response => response.json())
+            .then(json => {
+                debugger;
```
Comment (santiagolizardo) [inline src/components/projects/vault/VaultItemEdit.js:44]: Oops!

File: src/components/projects/vault/VaultItemEdit.js [hunk 2]
```diff
@@ -0,0 +1,127 @@
...
+                    setPassword(null);
+                }
+                else 
+                {
+                    var newItem = new Vault();
+                    newItem.name = json['name'];
+                    newItem.note = json['note'];
+                    newItem.value = json['value'];
+                    newItem.type = json['type'];
+                    newItem.reportable = json['reportable'];
```
Comment (santiagolizardo) [inline src/components/projects/vault/VaultItemEdit.js:57]: You could type less characters in this way:
```js
                    newItem.name = json.name;
                    newItem.note = json.note;
                    ....
```

File: src/components/projects/vault/VaultItemEdit.js (hunk 1 above)
Comment (gano3s) [reply src/components/projects/vault/VaultItemEdit.js:44]: Nice catch, thank you!

File: src/components/projects/vault/VaultItemEdit.js (hunk 2 above)
Comment (gano3s) [reply src/components/projects/vault/VaultItemEdit.js:57]: nice, I did not know this. Thank you!

**Pull request merged**